kotlinx-coroutines-core/native/src/MultithreadedDispatchers.kt

```diff
@@ -35,9 +35,24 @@ internal class WorkerDispatcher(name: String) : CloseableCoroutineDispatcher(),
     }
 
     override fun invokeOnTimeout(timeMillis: Long, block: Runnable, context: CoroutineContext): DisposableHandle {
-        // No API to cancel on timeout
-        worker.executeAfter(timeMillis.toMicrosSafe()) { block.run() }
-        return NonDisposableHandle
+        // Workers don't have an API to cancel sent "executeAfter" block, but we are trying
+        // to control the damage and reduce reachable objects by nulling out `block`
+        // that may retain a lot of references, and leaving only an empty shell after a timely disposal
+        class DisposableBlock(block: Runnable) : DisposableHandle, Function0<Unit> {
```

Comment:
It could be nice to highlight that this code being non-idiomatic is intentional.
```suggestion
        // This is a class and not an object with `block` in a closure because that would defeat the purpose.
        class DisposableBlock(block: Runnable) : DisposableHandle, Function0<Unit> {
```